### PR TITLE
LPS-205115 Update link on UserNotificationEvent to redirect user to System Settings.

### PR DESCRIPTION
**Pull request by thallescezar**
Related issue: https://liferay.atlassian.net/browse/LPS-205028

### CODE REVIEW

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:object

**Comment by thallescezar** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+				portletURLBuilderMockedStatic.verify(
+					() -> PortletURLBuilder.create(Mockito.any()));
+				Assert.assertTrue(true);
+			}
+		}
+	}
+
+	@Test
+	public void testGetLinkWhenNotificationIsRelatedToObjectEntries()
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java:142]: Rename to  `testGetLinkWhenObjectEntryHasNotDisplayPageFriendlyURL`

File: modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+		);
+
+		Assert.assertEquals(
+			"friendlyURL",
+			_objectUserNotificationsHandler.getLink(
+				_userNotificationEvent, _serviceContext));
+	}
+
+	@Test
+	public void testGetLinkWhenNotificationIsCreatedAfterGuestUserAddEntryThatExceedsLimits()
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java:84]: Rename this to `testGetLinkWhenGuestUserObjectEntrySubmissionsExceedsLimits`

File: modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+				Mockito.mockStatic(PortletURLBuilder.class)) {
+
+			portletURLBuilderMockedStatic.when(
+				() -> PortletURLBuilder.create(Mockito.any())
+			).thenReturn(
+				_portletURLStep
+			);
+
+			try (MockedStatic<PortalUtil> portalUtilMockedStatic =
+					Mockito.mockStatic(PortalUtil.class)) {
```
[the remark this answers is not in the document]
Comment (MarinhoFeliphe) [reply modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java:179]: Please do the same in the method above

File: modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+				).thenReturn(
+					_portletURL
+				);
+
+				_objectUserNotificationsHandler.getLink(
+					_userNotificationEvent, _serviceContext);
+
+				portletURLBuilderMockedStatic.verify(
+					() -> PortletURLBuilder.create(Mockito.any()));
+				Assert.assertTrue(true);
```
[the remark this answers is not in the document]
Comment (MarinhoFeliphe) [reply modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java:196]: Please do the same in the method above

File: modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+		);
+
+		Mockito.when(
+			_userNotificationEvent.getPayload()
+		).thenReturn(
+			"{ \"exceedsObjectEntryLimit\": true }"
+		);
+
+		try (MockedStatic<RequestBackedPortletURLFactoryUtil>
+				factoryUtilMockedStatic = Mockito.mockStatic(
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java:112]: Rename `factoryUtilMockedStatic` to `requestBackedPortletURLFactoryUtilMockedStatic`

Thanks!

File: modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+				Mockito.any(ThemeDisplay.class))
+		).thenReturn(
+			"friendlyURL"
+		);
+
+		Assert.assertEquals(
+			"friendlyURL",
+			_objectUserNotificationsHandler.getLink(
+				_userNotificationEvent, _serviceContext));
+	}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java:81]: Hey @thallescezar please apply my suggestion below:

```java
	public void testGetLinkWhenObjectEntryHasDisplayPageFriendlyURL()
		throws Exception {

		Mockito.when(
			_assetDisplayPageFriendlyURLProvider.getFriendlyURL(
				Mockito.any(InfoItemReference.class),
				Mockito.any(ThemeDisplay.class))
		).thenReturn(
			"http://localhost:8080/web/l/54321"
		);

		Assert.assertEquals(
			"http://localhost:8080/web/l/54321",
			_objectUserNotificationsHandler.getLink(
				_userNotificationEvent, _serviceContext));
	}
```

Thanks for covering this case as well!

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandler.java [hunk 7]
```diff
@@ -72,13 +76,33 @@ protected String getLink(
...
+		}
+
 		return PortletURLBuilder.create(
 			PortalUtil.getControlPanelPortletURL(
 				serviceContext.getRequest(), serviceContext.getScopeGroup(),
 				jsonObject.getString("portletId"), 0, 0,
 				PortletRequest.RENDER_PHASE)
 		).setMVCRenderCommandName(
-			"/object_entries/edit_object_entry"
+			"/configuration_admin/edit_configuration"
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandler.java:105]: Hey @thallescezar keep `"/object_entries/edit_object_entry"` here,

Thank you!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
Hey @thallescezar I've asked small changes, the most are related to unit tests where you've done a very nice job BTW, thanks for covering all cases of the class you changed I really appreciate this.

About the test try to create the verifications we've talked today.

Thanks for your hard work :tada:

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/liferay-objects/liferay-portal/pull/3055

Nice job @thallescezar!! :tada:

**Pull request closed without merging**